extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -438,7 +438,7 @@ async function hookUpDynamicOptions(context: WizardPageContext): Promise<void> {
 				const updateOptions = async () => {
 					const currentValue = await targetComponent.getValue();
 					if (field.dynamicOptions && field.options && fieldComponent && fieldComponent.setOptions) {
-						const targetValueFound = field.dynamicOptions.alternates.find(item => item.selection === currentValue);
+						const targetValueFound = field.dynamicOptions.alternates.find(item => item.selection === currentValue || (<azdata.CategoryValue>item.selection).name === currentValue);
```

Comment:
Make this into an if as well that checks the type of item.selection and if it's an object use name, otherwise just use item.selection directly